executor/simple.go

```diff
@@ -1180,7 +1181,36 @@ func (e *SimpleExec) executeCreateUser(ctx context.Context, s *ast.CreateUserStm
 		}
 
 		hostName := strings.ToLower(spec.User.Hostname)
-		sqlexec.MustFormatSQL(sql, valueTemplate, hostName, spec.User.Username, pwd, authPlugin, userAttributesStr, plOptions.lockAccount, recordTokenIssuer, plOptions.passwordExpired, plOptions.passwordLifetime)
+
+		var resQryHour, resUpdHour, resConHour, resUserCon int64 = 0, 0, 0, 0
+		for _, resOpt := range s.ResourceOptions {
```

Comment:
If currently these options are implemented only in the parser level, we should give warnings to users to mention it.

Reply:
We already do this since a3182eb84347a264dfa6872642ab5d7cac950fdf:
```
sql> CREATE USER 'foo'@'bar.example.com' IDENTIFIED BY 'foobar' WITH MAX_CONNECTIONS_PER_HOUR 1;
Query OK, 0 rows affected, 1 warning (0.0162 sec)
Warning (code 1064): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 90 near ""TiDB does not support WITH ConnectionOptions now, they would be parsed but ignored. 

sql> 
```

https://github.com/pingcap/tidb/blob/67e8f902f32fa793f45808bd7468fb6fe4bc5249/parser/parser.y#L13320